var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -296,9 +296,13 @@ def setup_build_environment(self, env):
         env.set('BUILD_BZIP2', 0)
         env.set('BZIP2_INCLUDE', spec['bzip2'].prefix.include)
         env.set('BZIP2_LIB', spec['bzip2'].libs.directories[0])
-        env.set('BUILD_ZLIB', 0)
-        env.set('ZLIB_INCLUDE', spec['zlib'].prefix.include)
-        env.set('ZLIB_LIB', spec['zlib'].libs.directories[0])
+        # spec['zlib'].libs.directories is/was not set when building mozjs
+        if 'zlib' in spec and spec['zlib'].libs.directories:
```

Comment:
zlib is always in the spec, I don't see what this changes

Reply:
spec['zlib'] is, and I know this is not to happen, but getting the lib directory below ysing`spec['zlib'].libs.directories[0]` triggered this:
```
Error: IndexError: list index out of range
var/spack/repos/builtin/packages/perl/package.py:289, in setup_build_environment:
  >>    289        env.set('ZLIB_LIB', spec['zlib'].libs.directories[0])
```
This part should likely be a separate bug report, I just thought I'd bring it to attention too.